pkg/lib/cockpit-components-dialog.jsx

```diff
@@ -327,3 +327,21 @@ export function show_modal_dialog(props, footerProps) {
 
     return dialogObj;
 }
+
+export function apply_modal_dialog(event) {
```

Comment:
That's not so much intuitive but I am happy you found a solution for dialog_open, can't think of anything simpler.

Reply:
Would it be better to call this `apply_current_modal_dialog`?

Reply:
I did not mean the function name :) I meant that's weird way of triggering the hander. aka trying to find the button with apply class and then clicking it virtually etc.

Reply:
Yeah, thatw as the easiest way to get cockpit-component-dialog to run a action callback in the right context, and it doesn't seem super wrong.